kits/apptentive-kit/src/main/java/com/mparticle/kits/ApptentiveKit.java

```diff
@@ -0,0 +1,243 @@
+	}
+
+	@Override
+	public List<ReportingMessage> logScreen(String screenName, Map<String, String> eventAttributes) throws Exception {
+		return null;
+	}
+
+	@Override
+	public List<ReportingMessage> logLtvIncrease(BigDecimal valueIncreased, String eventName, Map<String, String> contextInfo) {
+		return null;
+	}
+
+	@Override
+	public List<ReportingMessage> logEvent(CommerceEvent event) throws Exception {
+		if (!TextUtils.isEmpty(event.getProductAction())) {
+			Map<String, String> eventActionAttributes = new HashMap<String, String>();
+			CommerceEventUtil.extractActionAttributes(event, eventActionAttributes);
+
+			CommerceExtendedData apptentiveCommerceData = null;
+
+			TransactionAttributes transactionAttributes = event.getTransactionAttributes();
+			if (transactionAttributes != null) {
+				apptentiveCommerceData = new CommerceExtendedData();
+
+				String transaction_id = transactionAttributes.getId();
+				if (!TextUtils.isEmpty(transaction_id)) {
+					apptentiveCommerceData.setId(transaction_id);
+				}
+				Double transRevenue = transactionAttributes.getRevenue();
+				if (transRevenue != null) {
+					apptentiveCommerceData.setRevenue(transRevenue);
+				}
+				Double transShipping = transactionAttributes.getShipping();
+				if (transShipping != null) {
+					apptentiveCommerceData.setShipping(transShipping);
+				}
+				Double transTax = transactionAttributes.getTax();
+				if (transTax != null) {
+					apptentiveCommerceData.setTax(transTax);
+				}
+				String transAffiliation = transactionAttributes.getAffiliation();
+				if (!TextUtils.isEmpty(transAffiliation)) {
+					apptentiveCommerceData.setAffiliation(transAffiliation);
+				}
+				String transCurrency = eventActionAttributes.get(Constants.Commerce.ATT_ACTION_CURRENCY_CODE);
+				if (TextUtils.isEmpty(transCurrency)) {
+					transCurrency = Constants.Commerce.DEFAULT_CURRENCY_CODE;
+				}
+				apptentiveCommerceData.setCurrency(transCurrency);
+
+				// Add each item
+				List<Product> productList = event.getProducts();
+				if (productList != null) {
+					for (Product product : productList) {
+						CommerceExtendedData.Item item = new CommerceExtendedData.Item();
+						item.setId(product.getSku());
+						item.setName(product.getName());
+						item.setCategory(product.getCategory());
+						item.setPrice(product.getUnitPrice());
+						item.setQuantity(product.getQuantity());
+						item.setCurrency(transCurrency);
+						apptentiveCommerceData.addItem(item);
+					}
+				}
+			}
+
+			if (apptentiveCommerceData != null) {
+				Map<String, String> customData = event.getCustomAttributes();
+				Apptentive.engage(context, String.format("eCommerce - %s", event.getProductAction()),
+						Collections.<String, Object>unmodifiableMap(customData), apptentiveCommerceData);
+				List<ReportingMessage> messages = new LinkedList<ReportingMessage>();
+				messages.add(ReportingMessage.fromEvent(this, event));
+				return messages;
+			}
+
+		}
+		return null;
```

Comment:
Does Apptentive support the notion of user value/eCommerce purchases? It looks like it does not (Apptentive.engage us being called just as if this were a regular event) 

If that's the case, you can actually change  `ApptentiveKit`, removing the ECommerceForwarder interface implementation and this method entirely. The Android SDK will automatically see expand eCommerce events for you and will send them in as regular events, to the log event method above.